elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metric/ComputedMetric.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.metric;
+
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.datastores.aggregation.Schema;
+import com.yahoo.elide.datastores.aggregation.annotation.Meta;
+import com.yahoo.elide.datastores.aggregation.annotation.MetricComputation;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.StringJoiner;
+
+/**
+ * {@link Metric} annotated by {@link MetricComputation}.
+ */
+@Slf4j
+public class ComputedMetric implements Metric {
+
+    /**
+     * The placeholder reserved for constructing JQPL metricExpression, which requires a table prefix.
+     * <p>
+     * For example, consider the following entity with {@code expandingField = timeSpentPerSession or timeSpentPerGame}
+     * and {@code containingCls = FactTable.class}
+     * <pre>
+     * {@code
+     * public class FactTable {
+     *
+     *     {@literal @}MetricAggregation(sum.class)
+     *     Long sessions
+     *
+     *     {@literal @}MetricAggregation(sum.class)
+     *     Long timeSpent
+     *
+     *     {@literal @}MetricComputation(metricExpression = "timeSpent / sessions")
+     *     Float timeSpentPerSession
+     *
+     *     {@literal @}MetricComputation(metricExpression = "timeSpentPerSession / gameCount")
+     *     Float timeSpentPerGame
+     * }
+     * }
+     * </pre>
+     * One possible expanded JPQL metricExpression for {@code timeSpentPerSession} can be
+     * {@code com_oath_blah_FactTable.timeSpent / com_oath_blah_FactTable.sessions} and for {@code timeSpentPerGame} as
+     * {@code (com_oath_blah_FactTable.timeSpent/com_oath_blah_FactTable.sessions) / com_oath_blah_FactTable.gameCount}.
+     * {@link Schema} keeps such metricExpression in the form of {@code TABLE_PREFIX.timeSpent / TABLE_PREFIX.sessions}
+     * for {@code timeSpentPerSession} and
+     * {@code (TABLE_PREFIX.timeSpent/TABLE_PREFIX.sessions) / TABLE_PREFIX.gameCount} for
+     * {@code timeSpentPerGame}, where {@code TABLE_PREFIX} is specified by {@link #TABLE_PREFIX}.
+     * <p>
+     * <b>It is caller's responsibility to replace the {@link #TABLE_PREFIX} with the per-user specific value at
+     * runtime.</b>
+     */
+    public static final String TABLE_PREFIX = "<PREFIX>";
+
+    private static final long serialVersionUID = -5243438667754088900L;
+
+    @Getter
+    private final String name;
```

Comment:
A lot of these fields are common (with the concept of a Column described earlier).  We should move them to a common base class.